### DISCUSSION TITLE
Live theme reloading in open applications

### DISCUSSION THREAD

**Discussion by Cu3PO42** [tsujan/Kvantum#975] [Q&A]
Hi there! First I want to thank you for the tremendous work you've done with Kvantum.

I'd love to be able to change the theme of currently running Qt applications. My brain remembers this working before, but I can currently find no hard evidence of it at all.

I am currently running Kvantum 1.1 and tried changing the theme via both Kvantummanager's GUI and the CLI. As 'applications to be themed' I have tried both Qt5 and Qt6 applications. Both are themed correctly after an application restart, but not before.

I see you currently have a `restyleWindow` function in Kvantummanager. Might it be possible to execute something similar in other applications upon theme change? I suppose one could watch the Kvantum config for changes to be notified of that?

Unfortunately, I know virtually nothing about Qt or its internals, but if you believe this to be possible, I may take a crack at it.

**Comment by tsujan** [comment]
Being a style plugin, Kvantum can't apply its themes on the fly. Only a "platform theme" plugin has the permission to do such jobs (usually inside the directory `/usr/lib/qt6/plugins/platformthemes/`). KDE and LXQt have such plugins as well as config GUIs for them.

That being said, restarting apps (or logging out and in) is always better than an on-the-fly applying of styles/themes because Qt can't apply some details on the fly, especially those related to translucency.

Thanks for the kind words, BTW.

I'm converting this to a discussion for all users.

**Reply by Cu3PO42** [reply]
Thank you so much for your swift reply! I am currently using Qt5Ct/Qt6Ct as my platform them since I'm on neither Plasma nor LXQt. If I'm understanding you correctly, I would need it to force a theme reload. I took a peak and found it does support 
[watching the config file for changes](https://github.com/desktop-app/qt5ct/blob/master/src/qt5ct-qtplugin/qt5ctplatformtheme.cpp#L273) and will apply some settings in response to that. Although it does not appear to reload kvantum.

I suppose I'll need to get a much stronger graps on Qt internals before I can implement what I need. Thank you for your time!

> That being said, restarting apps (or logging out and in) is always better than an on-the-fly applying of styles/themes because Qt can't apply some details on the fly, especially those related to translucency.

This is a totally fair point and I appreciate the pointer to the limitations. However, I almost always only want to switch the colors in use by the application. I hope this might work (most of the time).
